modules/core/nuxeo-core-event/src/main/java/org/nuxeo/ecm/core/event/impl/EventServiceImpl.java

```diff
@@ -316,15 +316,20 @@ public void fireEvent(Event event) {
                 } else if (event.isBubbleException()) {
                     throw e;
                 } else if (event.isMarkedForRollBack()) {
+                    // rollback the transaction here
```

Comment:
Could you expand a bit maybe: make sure the transaction is marked rollback-only, even if some code later incorrectly swallows the rethrown exception